packages/core/src/ceramic.ts

```diff
@@ -488,15 +488,20 @@ class Ceramic implements CeramicApi {
     docId = normalizeDocID(docId)
     const docIdStr = docId.toString()
 
+    if (this._docmap[docIdStr]) {
+      return this._docmap[docIdStr]
+    }
+
     const doctypeHandler = this._doctypeHandlers[docId.typeName]
     if (!doctypeHandler) {
       throw new Error(docId.typeName + " is not a valid doctype")
     }
-
-    if (!this._docmap[docIdStr]) {
-      this._docmap[docIdStr] = await Document.load(docId, doctypeHandler, this.dispatcher, this.pinStore, this.context, opts)
+    const doc = await Document.load(docId, doctypeHandler, this.dispatcher, this.pinStore, this.context, opts)
+    if (!docId.version) {
+      // Only cache document if we're loading the baseId (i.e. the current version)
+      this._docmap[docIdStr] = doc
```

Comment:
In the future it would be nice if we could cache the state object, but not the instance of a doc.

Reply:
agreed